### PR TITLE
Model for saving Notification event audit info and Get request

### LINKED ISSUE #16: Create models for saving the status info of the notifications.
**Is your feature request related to a problem? Please describe.**
No

**Describe the solution you'd like**
New feature to save the audit information for notification requests

**Describe alternatives you've considered**

**Additional context**
Audit information helps to find status of the notification requests and analyze

### PR DESCRIPTION
**Pull request by akbhatta**
### Description
Model for saving Notification event audit info and Get request

[Tests]
Added Unit tests for models

Signed-off-by: @akbhatta

### Issues Resolved
#16 

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
  - [x] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: notifications/src/main/kotlin/org/opensearch/commons/notifications/action/GetNotificationEventRequest.kt [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+import org.opensearch.commons.utils.enumReader
+import org.opensearch.commons.utils.fieldIfNotNull
+import org.opensearch.commons.utils.logger
+import org.opensearch.search.sort.SortOrder
+import java.io.IOException
+
+/**
+ * Action Request for getting notification event.
+ */
+class GetNotificationEventRequest : ActionRequest, ToXContentObject {
```
[the remark this answers is not in the document]
Comment (akbhatta) [reply notifications/src/main/kotlin/org/opensearch/commons/notifications/action/GetNotificationEventRequest.kt:58]: Yes, that is right. but convention is to call it "get"

File: notifications/src/main/kotlin/org/opensearch/commons/notifications/action/GetNotificationEventRequest.kt [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+            var sortField: String? = null
+            var sortOrder: SortOrder? = null
+            var filterParams: Map<String, String> = mapOf()
+
+            XContentParserUtils.ensureExpectedToken(
+                XContentParser.Token.START_OBJECT,
+                parser.currentToken(),
+                parser
+            )
+            while (parser.nextToken() != XContentParser.Token.END_OBJECT) {
```
[the remark this answers is not in the document]
Comment (akbhatta) [reply notifications/src/main/kotlin/org/opensearch/commons/notifications/action/GetNotificationEventRequest.kt:93]: Makes sense. Thanks for the suggestion. Let me think about it.

File: notifications/src/main/kotlin/org/opensearch/commons/notifications/model/NotificationEventInfo.kt [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+data class NotificationEventInfo(
+    val eventId: String,
+    val lastUpdatedTime: Instant,
+    val createdTime: Instant,
+    val tenant: String,
+    val notificationEvent: NotificationEvent
+) : BaseModel {
+
+    init {
+        require(!Strings.isNullOrEmpty(eventId)) { "id is null or empty" }
```
[the remark this answers is not in the document]
Comment (akbhatta) [reply notifications/src/main/kotlin/org/opensearch/commons/notifications/model/NotificationEventInfo.kt:58]: Sure. will fix it.

[the remark this answers is not in the document]
Comment (akbhatta) [reply notifications/src/main/kotlin/org/opensearch/commons/notifications/model/NotificationEventInfo.kt]: fixed

**Review by jayeshathila — APPROVED** [review]

**Review by zhongnansu — APPROVED** [review]

**Pull request merged**